src/replicator.js

```diff
@@ -125,16 +125,34 @@ class Replicator {
     return replicator
   }
 
-  async start (rootstoreAddress, did, opts = {}) {
+  async start (rootstoreName, pubkey, did, opts = {}) {
+    if (this.rootstore) throw new Error('This method can only be called once before the replicator has started')
     this._did = did
     await this._joinPinningRoom(true)
-    this._publishDB({ odbAddress: rootstoreAddress })
 
-    this.rootstore = await this._orbitdb.feed(rootstoreAddress, ODB_STORE_OPTS)
+    const orbitStoreOpts = {
+      ...ODB_STORE_OPTS,
+      format: 'dag-pb'
+    }
+    orbitStoreOpts.accessController.write = [pubkey]
+    this.rootstore = await this._orbitdb.feed(rootstoreName, orbitStoreOpts)
+
+    this._pinningRoomFilter = null
+    this._publishDB({ odbAddress: this.rootstore.address.toString() })
     await this.rootstore.load()
-    this.rootstoreSyncDone = this.syncDB(this.rootstore)
-    const waitForSync = async () => {
+
+    try {
+      this.rootstoreSyncDone = this.syncDB(this.rootstore, { hasResponseTimeout: 2000 })
       await this.rootstoreSyncDone
+    } catch (err) {
+      // timeout error means there isn't a remote database answering HAS_ENTRIES requests,
+      // resolve without syncing
```

Comment:
Do I have it wrong that the pinning-node would resolve with  HAS_ENTRIES = 0 if the db isn't present?

Reply:
If that's not the case now, it would solve the problem if it did.

Reply:
you're right, it does - I think I wasn't seeing it before because of the `pinningRoomFilter` not being set correctly